dbt_meshify/main.py

```diff
@@ -444,17 +444,17 @@ def group(
 ) -> List[ChangeSet]:
     """
     Creates a new dbt group based on the selection syntax
-    Detects the edges of the group, makes their access public, and adds contracts to them
+    Detects the edges of the group, makes their access protected, and adds contracts to them
     """
 
     group_changes: List[ChangeSet] = ctx.forward(create_group)
 
     # Here's where things get a little weird. We only want to add contracts to projects
-    # that have a public access configuration on the boundary of our group. But, we don't
+    # that have a non-private access configuration on the boundary of our group. But, we don't
     # actually want to update our manifest yet. This puts us between a rock and a hard place.
     # to work around this, we can "trust" that create_group will always return a change for
-    # each public model, and rewrite our selection criteria to only select the models that
-    # will be having a public access config.
+    # each accessible model, and rewrite our selection criteria to only select the models that
+    # will be having a non-private access config.
 
     contract_changes = ctx.invoke(  # noqa: F841
```

Comment:
This brings up an interesting point! in my mind, contracts and public models are pretty synonymous, but i hadn't thought about a contract on a protected model. I suppose in the context of a single project, if these are indeed the interfaces between groups, this makes sense, since the effect of grouping a monolith should probably be the same as splitting a project, but I can anticipate some questions about this choice!

Reply:
Yeah, there will def be some questions!

IMHO, it all comes down to what the interfaces are. As you've stated, in a one-project (multi-hundred model and multi-AE team) world, the interfaces between groups ought to be contracted so that teams/groups can operate independently. But! We don't necessarily want to default to public access from other projects => protected models should be contracted when they're an interface.